src/instance.rs

```diff
@@ -214,6 +225,129 @@ unsafe impl<T> AsPyPointer for Py2<'_, T> {
     }
 }
 
+/// A borrowed equivalent to `Py2`.
+///
+/// The advantage of this over `&Py2` is that it avoids the need to have a pointer-to-pointer, as Py2
+/// is already a pointer to a PyObject.
+#[repr(transparent)]
+pub(crate) struct Py2Borrowed<'a, 'py, T>(
+    NonNull<ffi::PyObject>,
+    PhantomData<&'a Py<T>>,
+    Python<'py>,
+); // TODO is it useful to have the generic form?
```

Comment:
This `TODO` should go away because we are already using it for `PyBytes`.

Reply:
Agreed, I may also delete / comment away some of the other `Py2Borrowed` functionality not needed strictly for this PR. Hopefully can find time later tonight.